### DISCUSSION TITLE
Wrong Audio Track

### DISCUSSION THREAD

**Discussion by MTCMTC000** [aclap-dev/video-downloadhelper#332] [Help and Q&A]
### Your question

I am getting downloads with the wrong audio tracks off of YouTube.  The problem is random an rare.  This is the latest video to produce the error.  It has downloaded the Spanish language track twice.  The original English track is selected for the in page viewer but I suspect that is not relevant.  

https://www.youtube.com/watch?v=FLwpXJN2GC4

I cannot find anything about audio track selection for VDH.  Am I missing something?

### Addon & CoApp version and details

version: 8.2.2.8a1
channel: beta
build date: 2024-03-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): fcfd4d1b97e04bb7
platform: aarch64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:124.0) Gecko/20100101 Firefox/124.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "arm64",
    "node": "18"
  },
  "home": "/Users/mattcarter",
  "converterBinary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "n6.1-11-gc30f360955"
}

**Comment by jcfrog** [comment]
@paulrouget can you have a look at this one?

**Reply by MTCMTC000** [reply]
Did you forget an attachment or link?  I do not know what to look at.

**Comment by Mattmjs** [comment]
~~There are audio tracks for Spanish, Portuguese & English, click the gear button on the video and make sure English is selected.~~

Ok I see what's going on, for the master manifest there are several programs. Each program consists of three audio tracks then the video.

The Spanish audio track is always first in line followed by Portuguese then the English track. So it must automatically pick the Spanish
audio to go with the video. You could get the English by using ffmpeg itself or yt-dlp.

**Reply by MTCMTC000** [reply]
Unfortunately, I am not familiar with the inter-workings of the app to understand what you are referencing.  Can you give me a step-by-step instruction set?  More importantly, is this something that should be fixed in VDH?

**Reply by Mattmjs** [reply]
Try going in settings>More>Video qualities and uncheck the box for Ignore low quality videos. 
There should be an extra download available down the bottom of the menu.

**Reply by MTCMTC000** [reply]
It has been unchecked for longer than I can remember.  This is what is at the bottom of the list.
<img width="492" alt="VDH menu" src="https://github.com/aclap-dev/video-downloadhelper/assets/158776722/83ad578f-2067-43e1-be9e-bef67b02bf4e">

**Reply by Mattmjs** [reply]
That's it the webm down the bottom, it should have the english audio that your after.

**Reply by MTCMTC000** [reply]
Thank you.  I was looking for a different file format.  The download worked and with the English language track.  However, the resolution is so large, it will take forever on my current internet connection.  What's more, I would or will have to reformat it to a smaller file size.  Since this is proving to be a fluke I am not concerned about these issues.  I was afraid this was going to be a recurring problem.  So far, it has only happened once.

**Reply by Mattmjs** [reply]
I was wondering whether you wanted the 4k webm, but that is all it offers. You could get something smaller with english
but that would require using ffmpeg itself or yt-dlp. It might be too much of a hassle. If your interested I can show you how.

**Reply by MTCMTC000** [reply]
Interested, yes, but I hope I don't have to use it.  That said, I have been using VDH for more than a decade as I recall with this being a rare problem.  It is entirely your decision to post the instructions.  However, the instructions maybe something that could be turned into a specific instruction page for others to follow.

**Comment by MTCMTC000** [comment]
Using the file at the bottom of the menu provided a successful download.

**Comment by Mattmjs** [comment]
All you need to do is install **yt-dlp** and **ffmpeg**, which can be done through your software manager or in the terminal.
Once you have both, open the terminal and paste this for format list:

**yt-dlp -F https://www.youtube.com/watch?v=FLwpXJN2GC4**

[youtube] Extracting URL: https://www.youtube.com/watch?v=FLwpXJN2GC4
[youtube] FLwpXJN2GC4: Downloading webpage
[youtube] FLwpXJN2GC4: Downloading ios player API JSON
[youtube] FLwpXJN2GC4: Downloading android player API JSON
[youtube] FLwpXJN2GC4: Downloading android player API JSON
[youtube] FLwpXJN2GC4: Downloading android player API JSON
[youtube] FLwpXJN2GC4: Downloading android player API JSON
[youtube] FLwpXJN2GC4: Downloading player 3b96d06c
[youtube] FLwpXJN2GC4: Downloading m3u8 information
[info] Available formats for FLwpXJN2GC4:
ID    EXT   RESOLUTION FPS CH |   FILESIZE    TBR PROTO | VCODEC           VBR ACODEC      ABR ASR MORE INFO
-----------------------------------------------------------------------------------------------------------------------------------------------------
sb3   mhtml 48x27        0    |                   mhtml | images                                   storyboard
sb2   mhtml 80x45        0    |                   mhtml | images                                   storyboard
sb1   mhtml 160x90       0    |                   mhtml | images                                   storyboard
sb0   mhtml 320x180      0    |                   mhtml | images                                   storyboard
233-0 mp4   audio only        |                   m3u8  | audio only           unknown             [es-US] es-US - dubbed
233-1 mp4   audio only        |                   m3u8  | audio only           unknown             [pt-BR] Português (Brasil) - dubbed
233-2 mp4   audio only        |                   m3u8  | audio only           unknown             [en] English - original
234-0 mp4   audio only        |                   m3u8  | audio only           unknown             [es-US] es-US - dubbed
234-1 mp4   audio only        |                   m3u8  | audio only           unknown             [pt-BR] Português (Brasil) - dubbed
234-2 mp4   audio only        |                   m3u8  | audio only           unknown             [en] English - original
139-0 m4a   audio only      2 |    4.07MiB    49k https | audio only           mp4a.40.5   49k 22k [pt-BR] Portuguese (Brazil), low, m4a_dash
139-1 m4a   audio only      2 |    4.07MiB    49k https | audio only           mp4a.40.5   49k 22k [es-US] Spanish (United States), low, m4a_dash
139-2 m4a   audio only      2 |    4.07MiB    49k https | audio only           mp4a.40.5   49k 22k [en] English original (default), low, m4a_dash
249-0 webm  audio only      2 |    4.95MiB    59k https | audio only           opus        59k 48k [es-US] Spanish (United States), low, webm_dash
249-1 webm  audio only      2 |    4.99MiB    60k https | audio only           opus        60k 48k [pt-BR] Portuguese (Brazil), low, webm_dash
250-0 webm  audio only      2 |    6.39MiB    77k https | audio only           opus        77k 48k [es-US] Spanish (United States), low, webm_dash
250-1 webm  audio only      2 |    6.43MiB    77k https | audio only           opus        77k 48k [pt-BR] Portuguese (Brazil), low, webm_dash
249-2 webm  audio only      2 |    4.19MiB    50k https | audio only           opus        50k 48k [en] English original (default), low, webm_dash
250-2 webm  audio only      2 |    4.92MiB    59k https | audio only           opus        59k 48k [en] English original (default), low, webm_dash
140-0 m4a   audio only      2 |   10.81MiB   129k https | audio only           mp4a.40.2  129k 44k [pt-BR] Portuguese (Brazil), medium, m4a_dash
140-1 m4a   audio only      2 |   10.81MiB   129k https | audio only           mp4a.40.2  129k 44k [es-US] Spanish (United States), medium, m4a_dash
140-2 m4a   audio only      2 |   10.81MiB   129k https | audio only           mp4a.40.2  129k 44k [en] English original (default), medium, m4a_dash
251-0 webm  audio only      2 |   12.17MiB   146k https | audio only           opus       146k 48k [es-US] Spanish (United States), medium, webm_dash
251-1 webm  audio only      2 |   12.17MiB   146k https | audio only           opus       146k 48k [pt-BR] Portuguese (Brazil), medium, webm_dash
251-2 webm  audio only      2 |    8.73MiB   105k https | audio only           opus       105k 48k [en] English original (default), medium, webm_dash
602   mp4   256x144     12    | ~  7.78MiB    91k m3u8  | vp09.00.10.08    91k video only
269   mp4   256x144     24    | ~ 14.33MiB   168k m3u8  | avc1.4D400C     168k video only
160   mp4   256x144     24    |    5.03MiB    60k https | avc1.4D400C      60k video only          144p, mp4_dash
603   mp4   256x144     24    | ~ 12.74MiB   149k m3u8  | vp09.00.11.08   149k video only
278   webm  256x144     24    |    4.95MiB    59k https | vp09.00.11.08    59k video only          144p, webm_dash
229   mp4   426x240     24    | ~ 26.60MiB   311k m3u8  | avc1.4D4015     311k video only
133   mp4   426x240     24    |   11.37MiB   136k https | avc1.4D4015     136k video only          240p, mp4_dash
604   mp4   426x240     24    | ~ 21.78MiB   255k m3u8  | vp09.00.20.08   255k video only
242   webm  426x240     24    |    8.58MiB   103k https | vp09.00.20.08   103k video only          240p, webm_dash
230   mp4   640x360     24    | ~ 54.30MiB   635k m3u8  | avc1.4D401E     635k video only
134   mp4   640x360     24    |   21.48MiB   257k https | avc1.4D401E     257k video only          360p, mp4_dash
18    mp4   640x360     24  2 |   29.50MiB   353k https | avc1.42001E          mp4a.40.2       44k 360p
605   mp4   640x360     24    | ~ 39.37MiB   461k m3u8  | vp09.00.21.08   461k video only
243   webm  640x360     24    |   15.49MiB   186k https | vp09.00.21.08   186k video only          360p, webm_dash
231   mp4   854x480     24    | ~104.89MiB  1228k m3u8  | avc1.4D401E    1228k video only
135   mp4   854x480     24    |   44.67MiB   535k https | avc1.4D401E     535k video only          480p, mp4_dash
606   mp4   854x480     24    | ~ 57.50MiB   673k m3u8  | vp09.00.30.08   673k video only
244   webm  854x480     24    |   25.45MiB   305k https | vp09.00.30.08   305k video only          480p, webm_dash
22    mp4   1280x720    24  2 |  97.52MiB  1141k https | avc1.64001F          mp4a.40.2       44k 720p
232   mp4   1280x720    24    | ~198.62MiB  2324k m3u8  | avc1.64001F    2324k video only
136   mp4   1280x720    24    |   84.52MiB  1013k https | avc1.64001F    1013k video only          720p, mp4_dash
609   mp4   1280x720    24    | ~ 93.40MiB  1093k m3u8  | vp09.00.31.08  1093k video only
247   webm  1280x720    24    |   49.58MiB   594k https | vp09.00.31.08   594k video only          720p, webm_dash
270   mp4   1920x1080   24    | ~362.56MiB  4243k m3u8  | avc1.640028    4243k video only
137   mp4   1920x1080   24    |  160.80MiB  1927k https | avc1.640028    1927k video only          1080p, mp4_dash
614   mp4   1920x1080   24    | ~156.78MiB  1835k m3u8  | vp09.00.40.08  1835k video only
248   webm  1920x1080   24    |   82.74MiB   991k https | vp09.00.40.08   991k video only          1080p, webm_dash
620   mp4   2560x1440   24    | ~568.42MiB  6652k m3u8  | vp09.00.50.08  6652k video only
271   webm  2560x1440   24    |  327.84MiB  3928k https | vp09.00.50.08  3928k video only          1440p, webm_dash
625   mp4   3840x2160   24    | ~  1.19GiB 14316k m3u8  | vp09.00.50.08 14316k video only
313   webm  3840x2160   24    |  855.58MiB 10251k https | vp09.00.50.08 10251k video only          2160p, webm_dash

Let's say you want 480p video with english medium audio from that list:
135   mp4   854x480     24    |   44.67MiB   535k https | avc1.4D401E     535k video only          480p, mp4_dash
140-2 m4a   audio only      2 |   10.81MiB   129k https | audio only           mp4a.40.2  129k 44k [en] English original (default), medium, m4a_dash

Below is what you would put in terminal for the two formats:
**yt-dlp -f 135+140-2 https://www.youtube.com/watch?v=FLwpXJN2GC4**
